### PR TITLE
SSPROD-29308: Implement CloudAccount client v2 for Secure

### PR DESCRIPTION
**Pull request by ravinadhruve10**
Change summary:
-----------------
Adding a new Cloudauth Account client for making Secure BE (cloudauth) APIs to support API-only onboarding support.

### CODE REVIEW

**Review by nkraemer-sysdig — COMMENTED** [review]
Maybe we can use a `cloudauth` prefix instead of a `v2` suffix across the board? E.g. name the file `sysdig/internal/client/v2/cloudauth.go` and the constants `cloudauthAccountPath`/`cloudauthAccountsPath`?

Seems a bit odd to add `v2`s across the board, especially when refering to an endpoint with `v1` in it

File: sysdig/internal/client/v2/model.go [hunk 1]
```diff
@@ -411,6 +413,35 @@ type CloudAccountSecure struct {
...
+
+type CloudAccountFeatures struct {
+	SecureThreatDetection     *CloudAccountFeature
+	SecureConfigPosture       *CloudAccountFeature
+	SecureIdentityEntitlement *CloudAccountFeature
+	MonitorCloudMetrics       *CloudAccountFeature
+	SecureAgentlessScanning   *CloudAccountFeature
+}
+
+type CloudAccountSecureV2 struct {
```
Comment (ravinadhruve10) [inline sysdig/internal/client/v2/model.go:430]: Analyzing if we can import and use the Cloud Account object directly from `draios/protorepo/cloudauth`?
And then the `*FromResourceData` and `*ToResourceData` in resource*.go can do the from/to conversion of the fields as required.

**Comment by ravinadhruve10** [comment]
> Maybe we can use a `cloudauth` prefix instead of a `v2` suffix across the board? E.g. name the file `sysdig/internal/client/v2/cloudauth.go` and the constants `cloudauthAccountPath`/`cloudauthAccountsPath`?
> 
> Seems a bit odd to add `v2`s across the board, especially when refering to an endpoint with `v1` in it
> 
> > Yes, that works too and that's a fair point. I named it with v2 as per the framework suggestion, similar to how they did for alerts_v2.go. But cloudauth also works. (as long as we don't change the name of the service).

Done!

File: sysdig/internal/client/v2/model.go (hunk 1 above)
Comment (ravinadhruve10) [reply sysdig/internal/client/v2/model.go]: Done.

**Comment by ravinadhruve10** [comment]
All the latest changes are pulled and merged as part of https://github.com/sysdiglabs/terraform-provider-sysdig/pull/407 . Closing this PR then.

**Pull request closed without merging**